pkg/cmd/util/clientcmd/factory.go

```diff
@@ -375,8 +375,14 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			return kLogsForObjectFunc(object, options)
 		}
 	}
-	w.Printer = func(mapping *meta.RESTMapping, noHeaders, withNamespace, wide bool, showAll bool, showLabels, absoluteTimestamps bool, columnLabels []string) (kubectl.ResourcePrinter, error) {
-		return describe.NewHumanReadablePrinter(noHeaders, withNamespace, wide, showAll, showLabels, absoluteTimestamps, columnLabels), nil
+	w.Printer = func(mapping *meta.RESTMapping, options *kubectl.PrintOptions) (kubectl.ResourcePrinter, error) {
+		if mapping != nil {
```

Comment:
@rhcarvalho Thanks for all of your feedback!
> Why do we set options.KindName = mapping.Resource?!

Since this is the only file in the resource-printing process where we can access our own resource short forms from the same package where they are defined, I immediately assign the long resource name to the `PrintOptions` struct, just in case there is no alias for it:
```
options.KindName = mapping.Resource
```

Then, I look for our alias definition:

```
if alias, ok := resourceShortFormFor(mapping.Resource); ok {
	options.KindName = alias
}
```

If no alias is found, I simply leave the original long resource name. Then, once we jump back to `get.go` in the kubernetes package, it will see that we have already set a `KindName` from our list of resource aliases, and leave that as it is:

```
resourceName := resourcePrinter.GetResourceName() // Retrieves the name we set in the factory.go `Printer` method
if resourceName == "" {
	resourceName = mapping.Resource
}
```